src/constructions/direct_product.jl

```diff
@@ -0,0 +1,104 @@
+using Random
+using GroupsCore
+
+struct DirectProduct{Gt,Ht,GEl,HEl} <: GroupsCore.Group
```

Comment:
GroupsCore will not only be used by Oscar. These constructs at the Oscar level will be provided by GAP, and if you're looking at them from computations-in-groups-centric perspective -- are of little to no value to you. On the other hand if you want to use groups externally it's nice to have these constructs at hand. Note that `Constructions` submodule is not even exported.

I didn't add projections/embedings precisely because there was no discussion on them.

On the other hand I think computations with subgroups, etc should be provided by Oscar (a higher level functionality), not by a minimal package to define Group Interface.

btw. I wrote you last week on slack, maybe we could find some time to meet this week.